OpenRA.Game/World.cs

```diff
@@ -272,8 +275,8 @@ public void Remove(Actor a)
 				t.RemovedFromWorld(a);
 		}
 
-		public void Add(IEffect b) { effects.Add(b); }
-		public void Remove(IEffect b) { effects.Remove(b); }
+		public void Add(IEffect b) { effects.Add(b); if (b is ISync) syncedEffects.Add(b as ISync); }
+		public void Remove(IEffect b) { effects.Remove(b); if (b is ISync) syncedEffects.Remove(b as ISync); }
 		public void RemoveAll(Predicate<IEffect> predicate) { effects.RemoveAll(predicate); }
```

Comment:
No synced effect uses this predicate and I can't just add it as that won't compile.